extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlCreate.java

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import com.google.common.base.Strings;
+import io.kubernetes.client.Discovery;
+import io.kubernetes.client.common.KubernetesListObject;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.util.ModelMapper;
+import io.kubernetes.client.util.Namespaces;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.KubernetesApiResponse;
+import io.kubernetes.client.util.generic.options.CreateOptions;
+import java.util.Optional;
+import java.util.Set;
+
+public class KubectlCreate extends Kubectl.ApiClientBuilder<KubectlCreate>
+    implements Kubectl.Executable<KubernetesObject> {
+
+  KubectlCreate() {}
+
+  private KubernetesObject targetObj;
+
+  public KubectlCreate load(KubernetesObject obj) {
+    this.targetObj = obj;
+    return this;
+  }
+
+  @Override
+  public KubernetesObject execute() throws KubectlException {
+    Discovery discovery = new Discovery(apiClient);
+    try {
+      Set<Discovery.APIResource> apiResources = discovery.findAll();
```

Comment:
Do we want to do this every time? this could be pretty costly in the common case (e.g. Pods, Deployments, etc)

Perhaps only do this if the model mapper returns null?

Reply:
yeah we need an in-memory cache for api discovery, i added a TODO comment and will do it in a follow-up. WDYT?